test/integration-test/sql-parser/src/test/java/org/apache/shardingsphere/sql/parser/base/DynamicLoadingSQLParserParameterizedTest.java

```diff
@@ -81,12 +81,12 @@ protected static Collection<Map<String, String>> getResponse(final String sqlCas
         List<String> casesName = JsonPath.parse(caseContent).read("$..name");
         List<String> casesDownloadURL = JsonPath.parse(caseContent).read("$..download_url");
         List<String> casesHtmlURL = JsonPath.parse(caseContent).read("$..html_url");
+        List<String> casesType = JsonPath.parse(caseContent).read("$..type");
         IntStream.range(0, JsonPath.parse(caseContent).read("$.length()"))
                 .forEach(each -> {
-                    String eachName = casesName.get(each);
-                    if (eachName.endsWith(".sql") || eachName.endsWith(".test")) {
-                        result.add(ImmutableMap.of("name", eachName, "download_url", casesDownloadURL.get(each)));
-                    } else if (!eachName.contains(".")) {
+                    if (casesType.get(each).equals("file")) {
```

Comment:
this should be reversed. `"file".equals(casesType.get(each))`
it could avoid NPE

Reply:
fixed.